File: Spring/Gradle/firstproject/was-practice/src/main/java/org/example/calculator/logic/SubtractionOperator.java
package org.example.calculator.logic;

public class SubtractionOperator implements NewArithmeticOperator {

    @Override
    public boolean supports(String opearator) {
        return "-".equals(opearator);
    }

    @Override
    public int calculate(PositiveNumber operand1, PositiveNumber operand2) {
        return operand1.toInt() - operand2.toInt();
    }

}

File: Spring/Gradle/firstproject/servlet/src/main/java/org/example/calculator/tobe/SubtractionOperator.kt
package org.example.calculator.tobe

import org.example.calculator.domain.PositiveNumber

class SubtractionOperator : ArithmeticOperator {
    override fun supports(operator: String): Boolean {
        return "-" == operator
    }

    override fun calculate(operand1: PositiveNumber, operand2: PositiveNumber): Int {
        return operand1.toInt() - operand2.toInt()
    }
}